src/js/components/Main/Home/Home.container.jsx

```diff
@@ -9,13 +9,44 @@ const HomeContainer = props => {
   const [inputText, setInputText] = useState("");
   const [concerts, setConcerts] = useState([]);
   const [helpList, setHelpList] = useState([]);
+  const [recent, setRecent] = useState([]);
 
   useEffect(
       () => {
         props.getConcerts(setConcerts)
       },
       [props]
   )
+  useEffect(
+      () => {
+        findThreeNearest();
+      },
+      [concerts]
+  )
+
+
+  function findThreeNearest() {
+    if (concerts.length != 0) {
+      let sortedConcerts = concerts.sort((a, b) => {
+        if (Date.parse(a.date) < Date.parse(b.date))
+          return -1
+        else
+          return 1
+      })
+      let outArr = []
+      for (let i in sortedConcerts) {
+        if (Date.parse(sortedConcerts[i].date) > Date.now()) {
+          outArr.push(sortedConcerts[i - 0 + 2])
+          outArr.push(sortedConcerts[i - 0 + 1])
+          outArr.push(sortedConcerts[i - 0])
+          setRecent(outArr);
+          break;
+        }
+      }
+    }
+
+  }
+
```

Comment:
My solution for finding 3 the nearest concerts(i find translate mistake in "recent", will fix).